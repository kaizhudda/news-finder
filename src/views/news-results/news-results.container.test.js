import React from 'react';
import { mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import Summary from './index';

let wrapped;
const mockStore = configureMockStore([thunk]);
const intitalState = {
  news: { hits: [{ created_at: '2018-03-14T03:50:30.000Z', title: 'steph', url: '', author: 'test' }]  },
  filters: { tags: 'comment' },
  status: ''
}
const props = {
  hits: [{ created_at: '2018-03-14T03:50:30.000Z', title: 'steph', url: '', author: 'test' }],
  filters: { tags: 'comment' },
  status: '' 
}

describe('Summary', () => {
  beforeEach(() => {
    const store = mockStore(intitalState);
    wrapped = mount(<Summary store={store} {...props} />);
  });

  it('should render the Summary Container Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should contain filters prop', () => {
    expect(wrapped.props().filters).toBe(props.filters);
  });

  it('should contain hits prop', () => {
    expect(wrapped.props().hits).toBe(props.hits);
  });

  it('should contain filters prop', () => {
    expect(wrapped.props().status).toBe(props.status);
  });
});
