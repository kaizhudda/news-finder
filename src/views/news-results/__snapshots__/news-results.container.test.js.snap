// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should render the Summary Container Component correctly 1`] = `
<Connect(NewsResults)
  filters={
    Object {
      "tags": "comment",
    }
  }
  hits={
    Array [
      Object {
        "author": "test",
        "created_at": "2018-03-14T03:50:30.000Z",
        "title": "steph",
        "url": "",
      },
    ]
  }
  status=""
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <NewsResults
    fetchNewsData={[Function]}
    filters={
      Object {
        "tags": "comment",
      }
    }
    hits={
      Array [
        Object {
          "author": "test",
          "created_at": "2018-03-14T03:50:30.000Z",
          "title": "steph",
          "url": "",
        },
      ]
    }
    status=""
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div
      className="News-Results row center-lg center-md"
    >
      <div
        className="col-xs-12 col-sm-12 col-md-8 col-lg-7"
      >
        <div
          className="hit-item less-side-padding"
          key="0"
        >
          <span
            className="title"
          />
          <div
            className="support-hit"
          >
            <span>
              by 
              test
            </span>
            <span>
              a year ago
            </span>
          </div>
          <span
            className="comment"
          >
            <ReactMarkdown
              astPlugins={Array []}
              escapeHtml={false}
              parserOptions={Object {}}
              plugins={Array []}
              rawSourcePos={false}
              renderers={Object {}}
              skipHtml={false}
              sourcePos={false}
              transformLinkUri={[Function]}
            >
              <Root
                key="root-1-1"
              />
            </ReactMarkdown>
          </span>
        </div>
      </div>
    </div>
  </NewsResults>
</Connect(NewsResults)>
`;
